mne/decoding/receptive_field.py

```diff
@@ -0,0 +1,370 @@
+                                   newaxis=X.ndim)
+
+        # Remove timepoints that don't have lag data after delaying
+        X_del = X_del[self.keep_samples_]
+        y = y[self.keep_samples_]
+
+        # Convert to 2d by making epochs 1st axis and vstacking
+        X_del = X_del.reshape([-1, len(self.delays_) * n_feats], order='F')
+        y = y.reshape([-1, n_outputs], order='F')
+
+        # Update feature names if we have none
+        if self.feature_names is None:
+            self.feature_names = ['feature_%s' % ii for ii in range(n_feats)]
+        if len(self.feature_names) != n_feats:
+            raise ValueError('n_features in X does not match feature names '
+                             '(%s != %s)' % (n_feats, len(self.feature_names)))
+
+        self.estimator_.fit(X_del, y)
+
+        coefs = get_coef(self.estimator_, 'coef_')
+        coefs = coefs.reshape([n_outputs, n_feats, len(self.delays_)])
+        if len(coefs) == 1:
+            # Remove a singleton first dimension if only 1 output
+            coefs = coefs[0]
+        self.coef_ = coefs
+        return self
+
+    def predict(self, X, y=None):
+        """Generate predictions with a receptive field.
+
+        Parameters
+        ----------
+        X : array, shape (n_times[, n_epochs], n_channels)
+            The input features for the model.
+        y : None
+            Used for sklearn compatibility.
+
+        Returns
+        -------
+        y_pred : array, shape (n_times * n_epochs)
+            The output predictions with time concatenated.
+        """
+        if not hasattr(self, 'delays_'):
+            raise ValueError('Estimator has not been fit yet.')
+        X, y = self._check_dimensions(X, y, predict=True)
+        X_del = _delay_time_series(X, self.tmin, self.tmax, self.sfreq,
+                                   newaxis=X.ndim)
+        # Convert nans to 0 since sklearn will error otherwise
+        X_del[np.isnan(X_del)] = 0
+        X_del = X_del.reshape([-1, len(self.delays_) * X.shape[-1]], order='F')
+        y_pred = self.estimator_.predict(X_del)
+        return y_pred
+
+    def score(self, X, y):
+        """Score predictions generated with a receptive field.
+
+        This calls `self.predict`, then masks the output of this
+        and `y` with `self.mask_prediction_`. Finally, it passes
+        this to a `sklearn` scorer.
+
+        Parameters
+        ----------
+        X : array, shape (n_times[, n_epochs], n_channels)
+            The input features for the model.
+        y : array, shape (n_times[, n_epochs], n_outputs)
+            Used for sklearn compatibility.
+
+        Returns
+        -------
+        scores : list of float, shape (n_outputs,)
+            The scores estimated by ``scorer_`` for each output (e.g. mean
+            R2 of ``predict(X)``).
+        """
+        # Create our scoring object
+        self.scorer_ = _SCORERS[self.scoring]
```

Comment:
I don't believe we're using "scoring" objects under the hood at this point, only the public-facing API is sklearn style